### ISSUE TITLE
[GSSOC '23] Feature Request: Upcoming Features Command

### ISSUE THREAD

**Issue by slayerrr12** [roshan1337d/coding-contests-companion#31]
# Description:
As an open source contributor, I would like to propose the addition of a new command called **/upcoming-features** to provide information about the upcoming features of the coding contest retrieval bot. This feature will help users stay informed about the exciting new developments and foster engagement within the user community.

### Implementation Details:

1. Create a new command /upcoming-features in the bot's codebase.
2. Store upcoming features data in a JSON file or suitable data storage format.
3. Read the data from the JSON file and format it to display upcoming features to users.
4. Send a formatted message in the chat channel or via direct message to users containing the upcoming features list, descriptions, and expected release dates.
5. Handle cases where there are no upcoming features currently planned and display a corresponding message.
6. Optional: Provide additional features such as subscription to feature notifications and feedback submission.
7. Expected Behavior:
8. When users execute the /upcoming-features command, the bot should retrieve the upcoming features data and present it in a user-friendly format. Users should be able to see a list of upcoming features along with their descriptions and expected release dates.

### Additional Notes:

This feature will enhance user engagement by keeping them informed about the future plans and improvements of the bot.
The _JSON file_ or _data storage format_ should be easily maintainable and allow for easy addition or removal of upcoming features.
Please let me know if you need any further information or if there are any specific requirements for implementing this feature. I'm excited to contribute to the project and look forward to working on this feature.

**Comment by slayerrr12** [comment]
@roshan1337d  pls assign this issue to me as i have a experience with a similar codebase and working with discord.js i am too excited to make a fruitful contribution !

**Comment by roshan1337d** [comment]
Hello @slayerrr12 

I feel like those curious enough to know about upcoming features could just join in the demo/support server.

What would be nicer instead is a command that would allow me and other bot admins to send an announcement message to all the servers the bot is in to inform them about the new features after they are implemented.

Is that something you would be interested in doing?

**Comment by slayerrr12** [comment]
thanks for reviewing this....yes that could also be one way to go about it that there should be admin command through which we can send announcement for additional features to all guilds ....but i was thinking if the user wants to know about the those upcoming commands in that specific guild ... it would be handy to just type that command and get a look that those new commands that would be coming rather than joining the server..

**Comment by slayerrr12** [comment]
thanks for reviewing this....yes that could also be one way to go about it that there should be admin command through which we can send announcement for additional features to all guilds ....but i was thinking if the user wants to know about the those upcoming commands in that specific guild ... it would be handy to just type that command and get a look that those new commands that would be coming rather than joining the server..

**Comment by slayerrr12** [comment]
@roshan1337d the functionality of the bot ... should it be restricted to only retriving the contest related info or i can add some commands outside the domain as well but it the command be related to coding only.

**Comment by roshan1337d** [comment]
@slayerrr12 A command to know about upcoming features seems unnecessary to me. Not many people will use it and those who would want to can join the support server. It would also be another extra task to update the upcoming features.

An announcement command will be nice to have and I could assign you for it of you're interested.

About other coding related commands, what do you have in your mind?

**Comment by slayerrr12** [comment]
@roshan1337d Yes I am interested in implementing the announcement command feature pls assign it to me.


About the other coding related commands i was thinking about adding the insights of the prev contests amd features related to insights moreover i will be doing that in another issue

**Comment by roshan1337d** [comment]
@slayerrr12 go-ahead with the announcement command feature 👍

**Comment by roshan1337d** [comment]
@slayerrr12 it's been a while, do you want to continue working on this?

**Comment by slayerrr12** [comment]
i am working on this I have to dive a bit deeper in discord.js to actually Integrate this and make it work. I 'll update you within one week about the status of my work

**Comment by iamrudhresh** [comment]
Hey, hope you're doing great! As an GSSoC contributor, I'm keen on ramping up my involvement for GSSoC'24. Could you assign this issue to me? I'm excited to continue contributing to the project. Thanks a bunch!!
